### PR TITLE
Act on all services

### PR DESCRIPTION
**Pull request by antstorm**
Hi,

This is an extended version of #27, allowing to act on `--all` services when running `start`, `stop` or `restart`.

### CODE REVIEW

File: cmd/brew-services.rb [hunk 1]
```diff
@@ -121,20 +125,38 @@ def homebrew!
...
 
-    # Access current service
-    def service; @service ||= Service.new(Formula.factory(@formula)) if @formula end
+    # Access target services
+    def target_services
+      @target_services ||= begin
+        if @formula == '--all'
+          available_services
+        elsif @formula
+          Array(single_service)
```
Comment (mikemcquaid) [inline cmd/brew-services.rb]: `[single_service]` is a bit more idiomatic, I think.

File: cmd/brew-services.rb [hunk 2]
```diff
@@ -121,20 +125,38 @@ def homebrew!
...
+        end
+      end
+    end
+
+    # Access a single selected service
+    def single_service
+      @single_service ||= Service.new(Formula.factory(@formula)) unless @formula == '--all'
```
Comment (mikemcquaid) [inline cmd/brew-services.rb]: Maybe just `return nil if @formula == '--all'`?

File: cmd/brew-services.rb [hunk 3]
```diff
@@ -171,36 +193,32 @@ def run!
 
     # Check if formula has been found
     def check
-      odie("Formula missing, please provide a formula name") unless service
+      odie("Formula(s) missing, please provide a formula name or use --all") unless target_services
```
Comment (mikemcquaid) [inline cmd/brew-services.rb]: `Formula(s)` to `Formula(e)`

**Comment by mikemcquaid** [comment]
Looks pretty good to me but I want to do some more local testing. Thanks!

File: cmd/brew-services.rb (hunk 2 above)
Comment (antstorm) [inline cmd/brew-services.rb]: Which is exactly the same result, do you think it's more readable?

File: cmd/brew-services.rb (hunk 3 above)
Comment (antstorm) [inline cmd/brew-services.rb]: My bad, will fix that

**Comment by antstorm** [comment]
@mikemcquaid sure, feel free to test it. I did some testing myself and couldn't find any issues, but you are probably more aware of edge cases. I'll fix the other comments meanwhile.

File: cmd/brew-services.rb (hunk 2 above)
Comment (mikemcquaid) [inline cmd/brew-services.rb]: I think a little but up to you, I don't feel strongly.

File: cmd/brew-services.rb [hunk 4]
```diff
@@ -121,20 +125,39 @@ def homebrew!
       extend(FileUtils)
     end
 
-    # Access current service
-    def service; @service ||= Service.new(Formula.factory(@formula)) if @formula end
+    # Access target services
+    def target_services
+      @target_services ||= begin
+        if @formula == '--all'
```
Comment (mikemcquaid) [inline cmd/brew-services.rb]: Was thinking it might be nice to split `@formula == '--all'` logic everywhere into a method like `def all_services?` and also perhaps make use of `ARGV.include?` and/or `ARGV.delete` instead.

Comment (antstorm) [inline cmd/brew-services.rb]: That's a good idea!

**Comment by mikemcquaid** [comment]
Runs great locally, nice work. My only thought is it would be nice if `restart` restarted each service in order rather than stopping them all and then starting them all. Make sense?

**Comment by antstorm** [comment]
@mikemcquaid Thanks! I've added the sequential restart. It's a bit hacky, but works and is self-container. I think I know the way to improve this even further, but that's something for the next time.

File: cmd/brew-services.rb (hunk 4 above)
Comment (antstorm) [inline cmd/brew-services.rb]: Please take a look at the latest change and tell me if that's something that you had in mind :)

File: cmd/brew-services.rb [hunk 5]
```diff
@@ -152,55 +179,52 @@ def run!
...
+      else
+        onoe "Unknown command `#{@cmd}`"
+        usage(1)
       end
     end
 
     # Check if formula has been found
     def check
-      odie("Formula missing, please provide a formula name") unless service
+      odie("Formula(e) missing, please provide a fo rmula name or use --all") unless target_services
```
Comment (mikemcquaid) [inline cmd/brew-services.rb]: `fo rmula` to `formula`

Comment (antstorm) [inline cmd/brew-services.rb]: Somehow missed this one, fixed now

**Comment by antstorm** [comment]
@mikemcquaid I assume this one it good to go?

**Comment by mikemcquaid** [comment]
Yeh, sorry, been busy and need to get time for final review/testing here, thanks!

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by antstorm** [comment]
@mikemcquaid thanks mate, happy to be a part of it!
